hbase-client/src/main/java/org/apache/hadoop/hbase/security/HBaseSaslRpcClient.java

```diff
@@ -148,6 +148,16 @@ public boolean saslConnect(InputStream inS, OutputStream outS) throws IOExceptio
           inStream.readFully(saslToken);
         }
       }
+
+      try {
+        readStatus(inStream);
+      }
+      catch (IOException e){
+        if(e instanceof RemoteException){
```

Comment:
We still have this condition? I was talking about removing this condition because with this condition, we are swallowing `IOException` other than `RemoteException`, which we don't want to do.

Reply:
I think you removed the outer condition. Let me copy my previous comment:

I know `RemoteException` coming from `readStatus` is imp to us but we don't want to swallow IOException if `inStream.readInt()` throws one because `inStream.readInt()` is pre-requisite to know whether the status is `SaslStatus.SUCCESS`

I am talking about what is happening within `readStatus(inStream)` call.

Reply:
I just run a test on a test system. This time authentication was set up correctly and the goal was to see what happens in a normal usecase.
When it reached the new `readStatus(inStream);` The following exception was thrown:
```
java.net.SocketTimeoutException: 20000 millis timeout while waiting for channel to be ready for read. ch : java.nio.channels.SocketChannel[connected local=/xxx remote=yyy ]
        at org.apache.hadoop.net.SocketIOWithTimeout.doIO(SocketIOWithTimeout.java:164)
        at org.apache.hadoop.net.SocketInputStream.read(SocketInputStream.java:161)
        at org.apache.hadoop.net.SocketInputStream.read(SocketInputStream.java:131)
        at java.io.FilterInputStream.read(FilterInputStream.java:133)
        at java.io.BufferedInputStream.fill(BufferedInputStream.java:246)
        at java.io.BufferedInputStream.read(BufferedInputStream.java:265)
        at java.io.DataInputStream.readInt(DataInputStream.java:387)
        at org.apache.hadoop.hbase.security.HBaseSaslRpcClient.readStatus(HBaseSaslRpcClient.java:51)
```
This is an `IOException`. This is exactly the kind of exception that we do not want to re-throw, because it is caused by the additional `readStatus` and would not exist otherwise. The line `if(e instanceof RemoteException)` is there to filter out exceptions like this.

Reply:
Thanks @BukrosSzabolcs